sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2286,7 +2286,8 @@ case class MakeDate(
 
 // scalastyle:off line.size.limit
 @ExpressionDescription(
-  usage = "_FUNC_(year, month, day, hour, min, sec[, timezone]) - Create timestamp from year, month, day, hour, min, sec and timezone fields.",
+  usage = "_FUNC_(year, month, day, hour, min, sec[, timezone]) - Create timestamp from year, month, day, hour, min, sec and timezone fields. " +
+    "The result data type is consistent with the value of configuration `spark.sql.timestampType`",
```

Comment:
Please, use TIMESTAMP_TYPE.key. Using the SQL config has some benefits like:
- Easer to search in IDE by config name
- Refactoring. Maybe, we will rename it in the future. So, you will not need to modify this place.

Reply:
There will be errors:
```
annotation argument needs to be a constant; found: "_FUNC_(year, month, day, hour, min, sec[, timezone]) - Create timestamp from year, month, day, hour, min, sec and timezone fields. ".+(scala.StringContext.apply("The result data type is consistent ...
```